outfit.js

```diff
@@ -2,33 +2,35 @@ class Outfit {
   constructor(outfitFile) {
     this.title = outfitFile.title;
     this.id = outfitFile.id || this.generateIdNum();
-    this.garments = outfitFile.garments || [null, null, null];// NOTE: hats, clothes, accessories//
+    this.garments = outfitFile.garments || new Array(3).fill({garment: undefined, clickValue: undefined });
+    //NOTE: hats, clothes, accessories;
     this.background = outfitFile.background;
+
   }
-  
+
   changeGarment(garment) {
     if (garment.classList.contains('hats')) {
-      this.garments.splice(0, 1, garment.alt);
+      this.garments.splice(0, 1, {garment: garment.alt, clickValue: event.target.value});
     } else if (garment.classList.contains('clothes')) {
-      this.garments.splice(1, 1, garment.alt);
+      this.garments.splice(1, 1, {garment: garment.alt, clickValue: event.target.value});
     } else if (garment.classList.contains('accessories')) {
-      this.garments.splice(2, 1, garment.alt);
+      this.garments.splice(2, 1, {garment: garment.alt, clickValue: event.target.value});
     }
   }
 
   removeGarment(garment) {
     if (garment.classList.contains('hats')) {
-      this.garments.splice(0, 1);
+      this.garments.splice(0, 1, '');
     } else if (garment.classList.contains('clothes')) {
-      this.garments.splice(1, 1);
+      this.garments.splice(1, 1, '');
```

Comment:
i think the empty string needs to be replaced with {garment: undefined, clickValue: undefined

Reply:
Sounds great, we can for sure do that!